core/server/api/v2/utils/validators/input/posts.js

```diff
@@ -1,7 +1,25 @@
-const _ = require('lodash');
 const Promise = require('bluebird');
 const common = require('../../../../../lib/common');
 const utils = require('../../index');
+const jsonSchema = require('../utils/json-schema');
+
+const handleErrors = (errors) => {
```

Comment:
Was thinking if these schema validations should even be placed on this level or higher up for example closer to [here](https://github.com/TryGhost/Ghost/blob/master/core/server/api/shared/validators/handle.js#L53). 

Putting this handling on higher level would allow to process schema validation errors centrally and possibly having more generic way to include needed schemas, e.g would replace code like this:
```
const schema = require(`./schemas/posts-add`);
const definitions = require('./schemas/posts');
```
with something like:
```
const schema = require(`./schemas/${apiConfig.docName}-${apiConfig.method}`);
const definitions = require('./schemas/${apiConfig.docName}');
```

Reply:
Is "jsonSchema.validate" able to return full and custom errors?

Reply:
It returns an array of object with all validation errors at the moment (allErrors flag), woudl return just the first error without this flag. Sample error looks like following:
```
{
  "keyword":"required",
  "dataPath":".posts[0]",
  "schemaPath":"#/properties/posts/items/required",
  "params":{"missingProperty":"authors"},
  "message":"should have required property 'authors'"
}
```

Reply:
Why would it only just return one error?

e.g. inside of the jsonSchema.validate

```
if (errors)
   new errors.ValidationError({errorDetails: errors})
```

IMO it makes a lot of sense if the unit returns an error which is ready to return.

Reply:
Hm. Yeah, if we are not doing any more parsing to those errors outside the validator we could return, or maybe even `throw` :thinking:, an error straight away :+1: Should it still be a `ValidationError` or `BadRequestError` though?

Reply:
Also, what's your opinion on placing this validation not in validator itself but in proposed validation handler - https://github.com/TryGhost/Ghost/blob/master/core/server/api/shared/validators/handle.js#L53 ?

Reply:
IMO this is a ValidationError

Reply:
> Also, what's your opinion on placing this validation not in validator itself but in proposed validation handler - https://github.com/TryGhost/Ghost/blob/master/core/server/api/shared/validators/handle.js#L53 ?

Hm i would probably not over-optimise now? Not sure about it

Reply:
Ok. So, for now, will leave it as is. Next will be adding schema's for /tags so there will be more sense to reshufle things if needed.